Firecrawl credential validation POST lacks bounded timeout

### Self Checks

- [x] I have searched for existing issues and did not find an open report for Firecrawl credential validation timeouts.
- [x] I verified the behavior locally with a focused failing test before filing.

### Dify version

main branch

### Steps to reproduce

`services.auth.firecrawl.firecrawl.FirecrawlAuth.validate_credentials()` validates a Firecrawl API key by calling `_post_request()`, which delegates to `httpx.post()`:

```python
def _post_request(self, url, data, headers):
    return httpx.post(url, headers=headers, json=data)
```

There is no explicit timeout passed for the outbound credential-validation request.

I verified this with a focused unit-test assertion that the request should include a timeout:

```bash
uv run --project api pytest -o addopts='' api/tests/unit_tests/services/auth/test_firecrawl_auth.py::TestFirecrawlAuth::test_should_validate_valid_credentials_successfully -q
```

The test fails on current `main` because the `timeout` keyword is missing:

```text
E       KeyError: 'timeout'
```

### Expected Behavior

Firecrawl credential validation should use a bounded timeout, similar to the existing bounded timeout patterns used for other outbound validation/remote retrieval calls. A slow or hanging Firecrawl endpoint should not be able to block the worker indefinitely while users validate credentials.

### Actual Behavior

The Firecrawl auth provider calls `httpx.post(url, headers=headers, json=data)` without a timeout. If the remote endpoint stalls, credential validation relies on httpx defaults rather than an explicit project-level bound, unlike the newer timeout fixes in related auth/provider paths.

### Additional context

This is the Firecrawl auth-provider equivalent of the timeout hardening recently applied to nearby outbound provider calls. A minimal fix would define a small `httpx.Timeout` constant in `services/auth/firecrawl/firecrawl.py` and pass it to `_post_request()`.

--- a/api/services/auth/firecrawl/firecrawl.py
+++ b/api/services/auth/firecrawl/firecrawl.py
@@ -5,6 +5,10 @@ import httpx
 
 from services.auth.api_key_auth_base import ApiKeyAuthBase, AuthCredentials
 
+# Explicit bounded timeout for credential-validation requests so a slow or
+# hanging Firecrawl endpoint cannot block the worker indefinitely.
+_CREDENTIAL_TIMEOUT = httpx.Timeout(10.0)
+
 
 class FirecrawlAuth(ApiKeyAuthBase):
     def __init__(self, credentials: AuthCredentials):
@@ -42,7 +46,7 @@ class FirecrawlAuth(ApiKeyAuthBase):
         return f"{self.base_url.rstrip('/')}/{path.lstrip('/')}"
 
     def _post_request(self, url, data, headers):
-        return httpx.post(url, headers=headers, json=data)
+        return httpx.post(url, headers=headers, json=data, timeout=_CREDENTIAL_TIMEOUT)
 
     def _handle_error(self, response):
         try:

--- a/api/tests/unit_tests/services/auth/test_firecrawl_auth.py
+++ b/api/tests/unit_tests/services/auth/test_firecrawl_auth.py
@@ -86,6 +86,7 @@ class TestFirecrawlAuth:
             "https://api.firecrawl.dev/v1/crawl",
             headers={"Content-Type": "application/json", "Authorization": "Bearer test_api_key_123"},
             json=expected_data,
+            timeout=httpx.Timeout(10.0),
         )
 
     @pytest.mark.parametrize(

